### ISSUE TITLE
Figure out a good model to discourage cheating

### ISSUE THREAD

**Issue by idancali** [fluidtrends/carmel#94]
**Initial ideas**

1. don't allow people to take their own challenges - simple to do

2. create a machine learning algorithm that generates a "trust score" of some sort that raises "potential alarms" when it detects certain pattern - like a person taking the same course several times or taking a course of someone on their friends' list several times, plus a lot more things we can't even think of until we're in the midst of it

3. have people vote on other people's learning journeys and earn "trust" in the system based on how many votes you have - so if you believe Johnny is an awesome learner, upvote him and "encourage" him with your votes - but if you think someone might be cheating - downvote them

there's tons of other ideas but a mix of hard rules (like the inability to take your own challenges) + machine learning + human moderation - I think these could get us where we want to be as a community

Any other thoughts?

**Comment by bluejays3000** [comment]
“Endorsements” on LinkedIn is a good model to consider.

**Comment by idancali** [comment]
From **Shawn Lauzon**

- “verified” flag or upvotes to their completed challenge from people/companies who validated the person reproduced that challenge, they witnessed it

**Comment by bluejays3000** [comment]
I also like the ideas from 1) Amazon - Johnny is a verified purchaser of product xyz and 2) upwork (client reviews, general details and stats of recent completed work).

**Comment by shawnlauzon** [comment]
I think endorsements might be more useful on blockchain if there is a way to verify the endorser would actually have the ability to endorse. This is only the case if:

1. there is some relationship to the person being endorsed (otherwise how could they know?)
1. the endorser has some technical knowledge of the endorsed quality
1. that they don't have a history of blindly endorsing people (pseudo-blacklisting)

**Comment by shawnlauzon** [comment]
@bluejays3000 For carmel, what do you think would be similar to being a "verified purchaser"?

**Comment by shawnlauzon** [comment]
What needs to be validated is if the work is 1) original, and 2) from the person who did the work. The entire reason for the education system is credentializing (I think that's close to the word): Universities get paid big dollars to simply make sure there isn't cheating. If there was a way to (mostly) verify this, it would give confidence to employers.

I think a combination similar to this would give a lot of confidence:
1. Do all work in a special carmel editor (would note copy-paste from other sites, could do some analysis on typing authenticity)
2. webcam (idea from @bluejays3000) which matches the user and the person doing the typing.

**Comment by idancali** [comment]
So right now the way this is set to work is through the Carmel Desktop App. That means we have a safe + trusted + sandboxed work environment. Now just need to figure out the right algorithm

**Comment by cristianrosu76** [comment]
My thoughts about the endorsements are:
1. You create a challenge, you choose the skills that the challenge develop.
2. ONLY the people that are taking and completing your challenge can endorse you for the skills that the challenge targets. 
3. Once you reach a certain amount of endorsements for a skill, you get a certain number of XP.

**Comment by shawnlauzon** [comment]
@cristianrosu76 I'm not sure how completing a challenge qualifies one for endorsing someone. For example, I might be a super novice developer, create a crappy challenge and then another novice developer takes my challenge and endorses me.

**Comment by cristianrosu76** [comment]
@shawnlauzon I think that even if I'm  a beginner, if I were to create a challenge I should be incentivized to produce good materials that realy helps those that are taking my chalenge so they will be willing to endorse me for the skills that my challenge develop in them. If my challenge is a crap I don't know how many people would endorse me after taking my challenge and more than that I think they would flag my challenge as "poor". Makes sense?

**Comment by shawnlauzon** [comment]
@cristianrosu76 I agree that the challenge creator should be incentivized to produce good materials. However, endorsements (saying "I know this person has good skills") is not best done by someone who just learned the skills. Only someone who already has the skills can validate that another has the skills.

I do think that a new learner could rate the educational material. I just don't think they have the knowledge to rate the actual skills of the challenge creator.

**Comment by cristianrosu76** [comment]
@shawnlauzon I think you're right.
